web-backend/src/main/java/com/statusneo/vms/controller/VisitorController.java

```diff
@@ -108,9 +109,9 @@ public ResponseEntity<String> validateOtp(
             return ResponseEntity.badRequest().body("Invalid visit ID");
         }
 
-        boolean isValid = otpService.validateOtp(visit, otp);
+        VerificationResult result = otpService.validateOtp(visit, otp);
 
-        if (isValid) {
+        if (result.success()) {
             otpService.markVisitAsVerified(visit);
             return ResponseEntity.ok("<p class=\"text-green-600 font-bold\">OTP Verified Successfully!</p>");
```

Comment:
While you're changing and touching the UI, also move this HTML to a JTE fragment.